### DISCUSSION TITLE
Table Headers disappear

### DISCUSSION THREAD

**Discussion by bomshteyn** [awcodes/filament-table-repeater#80] [Q&A] [answered]
I have products table i only show the product selection for new rows by hiding the other fields unless product is selected.

This works great except when adding more than one row once you start adding a second row and on the headers get messed up.

### Adding First Product:
![Adding First Product](https://github.com/awcodes/filament-table-repeater/assets/4259699/75428ca9-3c15-4e92-99a7-e176c95d8b81)

### Once the first one has been added:
![Once the first one has been added](https://github.com/awcodes/filament-table-repeater/assets/4259699/cd43832c-2135-417f-b921-10a083bff4df)

### Adding second product:
![adding second product](https://github.com/awcodes/filament-table-repeater/assets/4259699/cb464bb0-df00-4a49-8f6b-bcac6334e8ef)

Notice how the headers disappear.

I tried adding the headers property with explicit list of headers, but still the same result

**Comment by awcodes** [comment, marked as answer]
You're breaking the table. Tables just don't work that way in html. You can't have per row hidden cells.

**Reply by bomshteyn** [reply]
Is there a way to pass to the cell to take up multiple cols as in colspan when it's empty?

**Reply by awcodes** [reply]
Currently, no.
